base/cvd/cuttlefish/host/commands/host_bugreport/main.cc

```diff
@@ -131,20 +137,30 @@ Result<void> CvdHostBugreportMain(int argc, char** argv) {
   CHECK(config) << "Unable to find the config";
 
   auto out_path = FLAGS_output.c_str();
-  std::unique_ptr<FILE, decltype(&fclose)> out(fopen(out_path, "wbe"), &fclose);
-  ZipWriter writer(out.get());
+  int err = 0;
+  std::unique_ptr<zip_t, decltype(&zip_discard)> archive(
+      zip_open(out_path, ZIP_CREATE | ZIP_TRUNCATE, &err), &zip_discard);
+  if (archive.get() == nullptr) {
+    zip_error_t error;
+    zip_error_init_with_code(&error, err);
+    std::string msg =
+        "zip_open failed: " + std::string(zip_error_strerror(&error));
+    zip_error_fini(&error);
+    return CF_ERR(msg);
+  }
 
-  auto save = [&writer, config](const std::string& path) {
-    SaveFile(writer, "cuttlefish_assembly/" + path, config->AssemblyPath(path));
+  auto save = [&archive, config](const std::string& path) {
+    SaveFile(archive.get(), "cuttlefish_assembly/" + path,
+             config->AssemblyPath(path));
```

Comment:
nit: having a lambda to simplify two function calls seems overkill

Reply:
Removed.